lms/models/register.py

```diff
@@ -0,0 +1,44 @@
+import re
+
+from flask import url_for
+from flask_babel import gettext as _  # type: ignore
+from flask_mail import Message  # type: ignore
+from itsdangerous import URLSafeTimedSerializer
+
+from lms.lmsdb.models import User
+from lms.lmsweb import config, webapp, webmail
+from lms.models.errors import EmptyPasswordError, UnhashedPasswordError
+
+
+SERIALIZER = URLSafeTimedSerializer(config.SECRET_KEY)
+
+
+def retrieve_salt(user: User) -> str:
+    try:
+        re.findall(r'\$(.*)\$', user.password)[0]
```

Comment:
Prefer to compile it and to give it a name. Let's go for a global variable of:

```
HASHED_PASSWORD = re.compile(r'^pdbkf2.+?\$(?P<salt>.+?)\$(?P<password>.+)')
```

That will compile the regex only once.

Then:
```
password = HASHED_PASSWORD.match(user.password)
salt = password.groupdict().get('salt')
```

(Again, maybe I'd move it to `Models/user.py`)